eidas-saml-parser/src/test/java/uk/gov/ida/notification/eidassaml/apprule/EspAuthnRequestAppRuleTest.java

```diff
@@ -41,6 +46,15 @@
 
     private EidasAuthnRequestBuilder request;
 
+    @ClassRule
+    public static final DropwizardClientRule metatronClientRule = createInitialisedClientRule(new TestMetatronResource());
+
+    @ClassRule
+    public static final EidasSamlParserAppRule eidasSamlParserAppRule = createEidasSamlParserRule(metatronClientRule);
+
+    @ClassRule
+    public static final RuleChain orderedRules = RuleChain.outerRule(metatronClientRule).around(eidasSamlParserAppRule);
+
```

Comment:
This felt better on the base class to me. The only configurable part of the setup is the test resource used for the Metatron stub, and we use the same thing in the two children of the base class. Keeping them on the base removes a bunch of duplication and means the `eidasSamlParserAppRule` doesn't need passing into so many calls to `postEidasAuthnRequest` unnecessarily.

Reply:
Keeping those things on the base class makes tests fail - they have to instantiated by each child independently, otherwise, they get killed by Dropwizard between class runs. This is the pattern we use for other AppRule tests (e.g. Gateway). They're static so they only need to be initialised once per class, no performance issue there.